scripts/pretraining/pretraining_utils.py

```diff
@@ -568,3 +570,42 @@ def dynamic_masking(self, input_ids, valid_lengths):
                                         masked_positions=masked_positions,
                                         masked_weights=masked_weights)
         return masked_input
+
+
+def get_electra_pretraining_model(model_name, ctx_l,
+                                  max_seq_length=128,
+                                  hidden_dropout_prob=0.1,
+                                  attention_dropout_prob=0.1,
+                                  generator_units_scale=None,
+                                  generator_layers_scale=None,
+                                  params_path=None):
+    """
+    A Electra Pretrain Model is built with a generator and a discriminator, in which
+    the generator has the same embedding as the discriminator but different backbone.
+    """
+    cfg, tokenizer, _, _ = get_pretrained_electra(
+        model_name, load_backbone=False)
+    cfg = ElectraModel.get_cfg().clone_merge(cfg)
+    cfg.defrost()
+    cfg.MODEL.hidden_dropout_prob = hidden_dropout_prob
+    cfg.MODEL.attention_dropout_prob = attention_dropout_prob
+    cfg.MODEL.max_length = max_seq_length
+    # Keep the original generator size if not designated
+    if generator_layers_scale:
+        cfg.MODEL.generator_layers_scale = generator_layers_scale
+    if generator_units_scale:
+        cfg.MODEL.generator_units_scale = generator_units_scale
+    cfg.freeze()
+
+    model = ElectraForPretrain(cfg,
+                               uniform_generator=False,
+                               tied_generator=False,
+                               tied_embeddings=True,
+                               disallow_correct=False,
+                               weight_initializer=TruncNorm(stdev=0.02))
+    if not params_path:
+        model.initialize(ctx=ctx_l)
+    else:
+        model.load_parameters(params_path, ctx=ctx_l)
```

Comment:
@sxjscience I'm currently running into the following error here:
```
AssertionError: Failed loading Parameter 'bias' from saved params: shape incompatible expected (256,) vs saved (192,)
```

It's not apparent to me if any other shape change happened somewhere. Do you have any idea?

Reply:
Turns out the generator scaler was missed.